src/contexts/TrackContextProvider.tsx

```diff
@@ -59,6 +64,27 @@ const TrackContextProvider = ({ children }: TrackContextProviderProps) => {
   const { deleteTrack } = useDeleteTrack();
   const { deleteMultiTracks } = useDeleteMultiTracks();
 
+  const [trackList, setTrackList] = useState<Track[]>([]);
+
+  const rawPage = getIntParam(QUERY_PARAM_KEYS.page);
+  const rawOrderBy = get(QUERY_PARAM_KEYS.orderBy);
+  const rawSortBy = get(QUERY_PARAM_KEYS.sortBy);
+  const rawSearchArtist = get(QUERY_PARAM_KEYS.searchArtist);
+
+  const page = O.getWithDefault(rawPage, INITIAL_QUERY_PARAMS_VALUE.page);
+  const orderBy = getValidatedOrDefaultQueryParam(rawOrderBy, isValidOrder, INITIAL_QUERY_PARAMS_VALUE.orderBy);
+  const sortBy = getValidatedOrDefaultQueryParam(
+    rawSortBy,
+    isTrackListSortableColumn,
+    INITIAL_QUERY_PARAMS_VALUE.sortBy,
+  );
+  const searchArtist = O.getWithDefault(rawSearchArtist, INITIAL_QUERY_PARAMS_VALUE.search);
+  const debouncedSearchArtist = useDebounce(searchArtist, 250);
+
+  const isValidPageParamInUrl = isValidQueryParam(rawPage, (page) => page >= INITIAL_QUERY_PARAMS_VALUE.page);
```

Comment:
maybe merge encapsulate `isValidQueryParam` in `useResetInvalidQueryParam` which can be renamed after changes `useValidatedQueryParam`. Looks like this util func and hook have high cohesion and can be merged into more efficient hook